src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * BMP388 Driver author: Dominic Clifton
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "build/build_config.h"
+
+#if defined(USE_BARO) && (defined(USE_BARO_BMP388) || defined(USE_BARO_SPI_BMP388))
```

Comment:
A better pattern than adding this complex check everywhere where we want to make sure that `USE_BARO_BMP388` is defined would be to add the following checks to `common_post.h`, and then only check for `#if defined(USE_BARO_BMP388)`:

```
#if !defined(USE_BARO)
#undef USE_BARO_BMP388
#endif 

#if !defined(USE_BARO_BMP388)
#undef USE_BARO_SPI_BMP388
#endif 
```

Reply:
good idea, will do.

Reply:
done

Reply:
as @MJ666 noted, this doesn't work since some targets only use the SPI variant.  Commit removed from PR.